### PR TITLE
TECSUPPORT-1349 [Hedera] User-friendly error message for network error

### PR DESCRIPTION
**Pull request by jackthta**
### 📝 Description

This PR will appease this error that some users are facing when trying to add an account:
- https://discord.com/channels/885256081289379850/922904577903317033/1004652978839490620
- https://discord.com/channels/885256081289379850/922904577903317033/1013769973627363389

The `AccountBalanceQuery` that happens during the synchronization step (in the add account flow) can fail if ...
1. ... a scheduled mainnet update is happening
2. ... nodes are unavailable to process the request due to the load of transactions that are being processed

Before this PR, an ambiguous error message would get thrown at the user (shown in the discord links). This PR will catch that potential failure and show a more user friendly error message.

### ❓ Context

- **Impacted projects**: `Live Common`, `Live Desktop`, and `Live Mobile` 
- **Linked resource(s)**: https://ledgerhq.atlassian.net/browse/TECSUPPORT-1349 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

**Live Desktop**
- Screen Shot 2022-09-12 at 09 47 42

**Live Mobile**
- Screen Shot 2022-09-12 at 10 40 51

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by haammar-ledger — APPROVED** [review]

File: libs/ledger-live-common/src/families/hedera/api/network.ts [hunk 1]
```diff
@@ -37,9 +37,15 @@ export async function getAccountBalance(
...
-    accountId,
-  }).execute(getClient());
+  let accountBalance;
+
+  try {
+    accountBalance = await new hedera.AccountBalanceQuery({
+      accountId,
+    }).execute(getClient());
+  } catch {
+    throw Error("Your account has not been added. Please try again.");
```
Comment (gre) [inline libs/ledger-live-common/src/families/hedera/api/network.ts:47]: we must create a new error so we can later translate it on the LLD / LLM side

**Review by gre — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 2]
```diff
@@ -5036,6 +5036,9 @@
     },
     "StellarSourceHasMultiSign": {
       "title": "Please disable multisign to make Stellar transactions"
+    },
+    "HederaAddAccountError": {
+      "title": "Your account has not been added. Please try again."
     }
```
Comment (haammar-ledger) [inline apps/ledger-live-desktop/static/i18n/en/app.json:5042]: Perfect :+1: But can you please do the same for Live Mobile?
https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-mobile/src/locales/en/common.json

File: libs/ledger-live-common/src/families/hedera/api/network.ts [hunk 3]
```diff
@@ -37,9 +38,15 @@ export async function getAccountBalance(
...
-    accountId,
-  }).execute(getClient());
+  let accountBalance;
+
+  try {
+    accountBalance = await new hedera.AccountBalanceQuery({
+      accountId,
+    }).execute(getClient());
+  } catch {
+    throw new HederaAddAccountError()
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/hedera/api/network.ts:48]: Missing semicolon here, prettier is complaining :slightly_smiling_face: 
https://github.com/LedgerHQ/ledger-live/actions/runs/3137430722/jobs/5095587144#step:10:2848

Protip: you can setup your IDE to automatically apply prettier formatting every time you save a file.

File: libs/ledger-live-common/src/families/hedera/errors.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { createCustomErrorClass } from "@ledgerhq/errors";
+
+export const HederaAddAccountError = createCustomErrorClass("HederaAddAccountError");
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/hedera/errors.ts:3]: Prettier issue here as well it seems
https://github.com/LedgerHQ/ledger-live/actions/runs/3137430722/jobs/5095587144#step:10:2852

**Review by haammar-ledger — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/static/i18n/en/app.json (hunk 2 above)
Comment (jackthta) [reply apps/ledger-live-desktop/static/i18n/en/app.json:5042]: Oh yes definitely, sorry about that haha.

File: libs/ledger-live-common/src/families/hedera/api/network.ts (hunk 3 above)
Comment (jackthta) [reply libs/ledger-live-common/src/families/hedera/api/network.ts]: Thanks for the tip! I should really put that in habit haha

**Review by gre — APPROVED** [review]

**Review by haammar-ledger — APPROVED** [review]

**Pull request merged**